torch/csrc/jit/script/concrete_module_type.h

```diff
@@ -41,83 +43,24 @@ enum class IterableModuleKind { NONE, LIST, DICT };
 // share a JIT type (and thus a ConcreteModuleType), then they behave the same
 // way when you access attributes on them.
-class VISIBILITY_HIDDEN ConcreteModuleType {
- public:
-  // ConcreteModuleType has two states.
-  // 1. Building: First we build it up, during the ScriptModule conversion
-  // process
-  //      ... to transition, we freeze the type by associating with a JIT type.
-  // 2. Querying: Then we ask use it as a source of truth during method
-  // compilation.
-  //              During this time the ModuleType is effectively const.
-  // Yes, it could be two different types. Not terribly worth the verbosity.
-
-  /**
-   * Builder methods (jitType must be null)
-   */
-  void addPyClass(py::object pyClass);
-  void addConstant(std::string name, py::object value);
-  void addAttribute(std::string name, TypePtr type, bool isParameter);
-  void addFunctionAttribute(
-      std::string name,
-      const TypePtr& type,
-      py::object pyFunction);
-
-  // add a submodule to the ConcreteModuleType can either be ConcreteModuleType
-  // that get constructed recursively, or InterfaceType (Module)
-  void addModule(std::string name, std::shared_ptr<ConcreteModuleType> meta);
-  void addModuleInterface(std::string name, const TypePtr& type);
-
-  void addOverload(
-      std::string methodName,
-      std::vector<std::string> overloadedMethodNames);
-  void addFailedAttribute(std::string name, std::string failureReason);
-  void setIterableModuleKind(IterableModuleKind kind);
-  void setPoisoned();
-
-  /**
-   * Freezing methods
-   */
-
-  // Based on the data in this ConcreteType, create an equivalent JIT type and
-  // associate this module type with it.
-  ClassTypePtr createNewTypeFromThis();
-  // Associate the provided type with this ConcreteType
-  void addJitType(ClassTypePtr type);
-
-  /**
-   * Query methods (jitType must be non-null)
-   */
-  ClassTypePtr getJitType() const;
-  py::object getPyClass() const;
-  IterableModuleKind getIterableModuleKind() const;
-  c10::optional<py::object> findConstant(const std::string& name) const;
-  c10::optional<std::vector<std::string>> findOverloads(
-      const std::string& name) const;
-  c10::optional<Function*> findFunctionAttribute(const std::string& name) const;
-  std::shared_ptr<ConcreteModuleType> findSubmoduleConcreteType(
-      const std::string& name) const;
-  c10::optional<std::string> findFailedAttribute(const std::string& name) const;
-
-  // These getters are only here to return things as types that can be
-  // automatically converted by pybind.
-  std::unordered_map<std::string, py::object> getConstantsPy() const;
-  std::unordered_map<std::string, std::pair<TypePtr, bool>> getAttributesPy()
-      const;
-  std::vector<std::pair<std::string, TypePtr>> getModulesPy() const;
 
+// ConcreteModuleType has two phases.
+// 1. Creation: First we build it up, during the ScriptModule conversion
+// process. This is represented by RawConcreteModuleType.
```

Comment:
Before I found this comment, but after reading the diff I could not figure out which of the two types was which. Can we call RawConcreteModuleType ConcreteModuleTypeBuilder?

Reply:
lol that was my first name for it 😛